lib/quicly.c

```diff
@@ -1426,11 +1426,15 @@ static int record_pn(quicly_ranges_t *ranges, uint64_t pn, int *is_out_of_order)
     return 0;
 }
 
-static int record_receipt(struct st_quicly_pn_space_t *space, uint64_t pn, int is_ack_only, int64_t now, int64_t *send_ack_at)
+static int record_receipt(struct st_quicly_pn_space_t *space, uint64_t pn, int is_ack_only, int64_t now, int64_t *send_ack_at,
+                          uint64_t *received_out_of_order)
 {
     int ret, ack_now, is_out_of_order;
 
-    if ((ret = record_pn(&space->ack_queue, pn, &is_out_of_order)) != 0)
+    ret = record_pn(&space->ack_queue, pn, &is_out_of_order);
+    if (is_out_of_order)
+        *received_out_of_order += 1;
+    if (ret != 0)
```

Comment:
How about updating `*received_out_of_order` only when `ret` is zero?

While we can rely on `is_out_of_order` returning the correct value even when `ret` indicates an error, I do not think we need to handle that case of having lines of code that sticks out.

Reply:
Makes sense, thank you for the suggestion, I've adapted the PR.